### PR TITLE
Add optional warning panel to questions

### PR DESCRIPTION
**Pull request by ajmaddaford**
### What is the context of this PR?

Adds the new optional warning panel to questions.

### How to review 

Use the `test_list_collector` questionnaire to check that the feature works. 

### Checklist

* [-] New static content marked up for translation
* [x] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: tests/integration/questionnaire/test_questionnaire_list_collector.py [hunk 1]
```diff
@@ -259,3 +259,12 @@ def test_cancel_text_displayed_on_edit_block_if_exists(self):
         self.get(change_link)
 
         self.assertInBody("Don’t need to change anything?")
+
+    def test_warning_text_displayed_on_remove_block_if_exists(self):
```
Comment (LJBabbage) [inline tests/integration/questionnaire/test_questionnaire_list_collector.py:263]: I wonder if we need the "if_exists"? I don't believe you can even get to the remove block if the list item doesn't exists.

File: templates/partials/question.html [hunk 2]
```diff
@@ -16,10 +15,20 @@
   {%- endif -%}
 {%- endset -%}
 
+{% set question_warning %}
+  {%- if question.warning -%}
+    {% call onsPanel({
+      "id": "question-warning-" + question.id,
```
Comment (MebinAbraham) [inline templates/partials/question.html:21]: The recommended way to concatenate string in Jinja is using a tilde. I know we use `+` in a lot of places but for new code, we should keep it as recommended.
```suggestion
      "id": "question-warning-" ~ question.id,
```

File: scripts/run_validator.sh [hunk 3]
```diff
@@ -1,4 +1,4 @@
 #!/usr/bin/env bash
-tag=latest
+tag=add-question-warning
```
Comment (MebinAbraham) [inline scripts/run_validator.sh:2]: Reminder to revert
```suggestion
tag=latest
```

File: tests/integration/questionnaire/test_questionnaire_list_collector.py (hunk 1 above)
Comment (ajmaddaford) [reply tests/integration/questionnaire/test_questionnaire_list_collector.py:263]: This was copied from the preceding tests. I read it as meaning "if the 'warning' property exists".

File: tests/integration/questionnaire/test_questionnaire_list_collector.py [hunk 4]
```diff
@@ -259,3 +259,12 @@ def test_cancel_text_displayed_on_edit_block_if_exists(self):
...
+        self.launchSurvey("test_list_collector")
+        self.post(action="start_questionnaire")
+        self.post({"anyone-else": "Yes"})
+        self.add_person("Someone", "Else")
+        remove_link = self.get_link("1", "Remove")
+        self.get(remove_link)
+        self.assertInBody("All of the information about this person will be deleted")
```
[the remark this answers is not in the document]
Comment (ajmaddaford) [reply tests/integration/questionnaire/test_questionnaire_list_collector.py:270]: Possibly, this was copied from the previous test. I'm not sure we consistently check for ids in any of our tests.

File: templates/partials/question.html (hunk 2 above)
Comment (ajmaddaford) [reply templates/partials/question.html:21]: Ok. I have updated the other occurrence of this in this file as well.

File: tests/integration/questionnaire/test_questionnaire_list_collector.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (LJBabbage) [reply tests/integration/questionnaire/test_questionnaire_list_collector.py:270]: The ID would come from the panel, meaning existence of a warning panel compared to just isolated text, and therefore more reassurance of correct formatting. As its also the question id you can check the warning and question go together. However saying that, I guess its not fundamentally needed, so shouldn't block

[the remark this answers is not in the document]
Comment (ajmaddaford) [reply tests/integration/questionnaire/test_questionnaire_list_collector.py:273]: I've added a test for the panel.

**Pull request merged**